test/Microsoft.ML.Tests/TrainerEstimators/TreeEnsembleFeaturizerTest.cs

```diff
@@ -108,5 +115,700 @@ public void TreeEnsembleFeaturizerOutputSchemaTest()
+            var secondMetrics = ML.Regression.Evaluate(secondPrediction);
+
+            // The second pipeline trains a tree featurizer on a bin-based normalized feature, so the second pipeline
+            // is different from the first pipeline.
+            Assert.NotEqual(metrics.MeanAbsoluteError, secondMetrics.MeanAbsoluteError);
+            Assert.NotEqual(metrics.MeanSquaredError, secondMetrics.MeanSquaredError);
+        }
+
+        [Fact]
+        public void TestFastTreeBinaryFeaturizationInPipelineWithOptionalOutputs()
+        {
+            int dataPointCount = 200;
+            var data = SamplesUtils.DatasetUtils.GenerateBinaryLabelFloatFeatureVectorFloatWeightSamples(dataPointCount).ToList();
+            var dataView = ML.Data.LoadFromEnumerable(data);
+            dataView = ML.Data.Cache(dataView);
+
+            var trainerOptions = new FastTreeBinaryTrainer.Options
+            {
+                NumberOfThreads = 1,
+                NumberOfTrees = 10,
+                NumberOfLeaves = 4,
+                MinimumExampleCountPerLeaf = 10,
+                FeatureColumnName = "Features",
+                LabelColumnName = "Label"
+            };
+
+            var options = new FastTreeBinaryFeaturizationEstimator.Options()
+            {
+                InputColumnName = "Features",
+                TrainerOptions = trainerOptions,
+                TreesColumnName = null,
+                PathsColumnName = null,
+                LeavesColumnName = "Leaves"
+            };
+
+
+            bool isWrong = false;
+            try
+            {
+                var wrongPipeline = ML.Transforms.FeaturizeByFastTreeBinary(options)
+                    .Append(ML.Transforms.Concatenate("CombinedFeatures", "Features", "Trees", "Leaves", "Paths"))
+                    .Append(ML.BinaryClassification.Trainers.SdcaLogisticRegression("Label", "CombinedFeatures"));
+                var wrongModel = wrongPipeline.Fit(dataView);
+            }
+            catch
+            {
+                isWrong = true; // Only "Leaves" is produced by the tree featurizer, so accessing "Trees" and "Paths" will lead to an error.
+            }
+            Assert.True(isWrong);
+
+            var pipeline = ML.Transforms.FeaturizeByFastTreeBinary(options)
+                .Append(ML.Transforms.Concatenate("CombinedFeatures", "Features", "Leaves"))
+                .Append(ML.BinaryClassification.Trainers.SdcaLogisticRegression("Label", "CombinedFeatures"));
+            var model = pipeline.Fit(dataView);
+            var prediction = model.Transform(dataView);
+            var metrics = ML.BinaryClassification.Evaluate(prediction);
+
+            Assert.True(metrics.Accuracy > 0.98);
+            Assert.True(metrics.LogLoss < 0.05);
+            Assert.True(metrics.AreaUnderPrecisionRecallCurve > 0.98);
+        }
+
+        [Fact]
+        public void TreeEnsembleFeaturizingPipelineMulticlass()
```

Comment:
@daholste and I were able to map the `Key` to a `float` using a `CustomMapping()`.

I'd recommend the multiclass unit test be:
```suggestion
        [Fact]
        public void TreeEnsembleFeaturizingPipelineMulticlass()
        {
            int dataPointCount = 1000;
            var data = SamplesUtils.DatasetUtils.GenerateRandomMulticlassClassificationExamples(dataPointCount).ToList();
            var dataView = ML.Data.LoadFromEnumerable(data);
            dataView = ML.Data.Cache(dataView);
 
            var trainerOptions = new FastForestRegressionTrainer.Options
            {
                NumberOfThreads = 1,
                NumberOfTrees = 10,
                NumberOfLeaves = 4,
                MinimumExampleCountPerLeaf = 10,
                FeatureColumnName = "Features",
                LabelColumnName = "FloatLabel",
                ShuffleLabels = true
            };
           
            var options = new FastForestRegressionFeaturizationEstimator.Options()
            {
                InputColumnName = "Features",
                TreesColumnName = "Trees",
                LeavesColumnName = "Leaves",
                PathsColumnName = "Paths",
                TrainerOptions = trainerOptions
            };
 
            Action<RowWithKey, RowWithFloat> actionConvertKeyToFloat = (RowWithKey rowWithKey, RowWithFloat rowWithFloat) =>
            {
                rowWithFloat.FloatLabel = rowWithKey.KeyLabel == 0 ? float.NaN : rowWithKey.KeyLabel - 1;
            };
 
            var split = ML.Data.TrainTestSplit(dataView, 0.5);
            var trainData = split.TrainSet;
            var testData = split.TestSet;
 
            var pipeline = ML.Transforms.Conversion.MapValueToKey("KeyLabel", "Label")
                .Append(ML.Transforms.CustomMapping(actionConvertKeyToFloat, "KeyLabel"))
                .Append(ML.Transforms.FeaturizeByFastForestRegression(options))
                .Append(ML.Transforms.Concatenate("CombinedFeatures", "Trees", "Leaves", "Paths"))
                .Append(ML.MulticlassClassification.Trainers.SdcaMaximumEntropy("KeyLabel", "CombinedFeatures"));
 
            var model = pipeline.Fit(trainData);
            var prediction = model.Transform(testData);
            var metrics = ML.MulticlassClassification.Evaluate(prediction, labelColumnName: "KeyLabel");
 
            Assert.True(metrics.MacroAccuracy > 0.6);
            Assert.True(metrics.MicroAccuracy > 0.6);
       }
       
        class RowWithKey
        {
            [KeyType()]
            public uint KeyLabel { get; set; }
        }
       
        class RowWithFloat
        {
            public float FloatLabel { get; set; }
        }
```

Specifically, this is using a `CustomMapping()` to convert the `Key` to a `float` for use in the TreeFeat's FastForest regression. The current method in the unit test requires a user to know/list all of the values in their Label (and their type). The `CustomMapping()` style is easier for a user to replicate for their dataset.

We also added a `TrainTestSplit()` so we're not testing on the training set, and we removed the original features from the `Concatenate()` to ensure the TreeFeat's output features are useful.